src/index.ts

```diff
@@ -349,10 +350,13 @@ export class CruxClient extends CruxPayPeer {
 
     public putAddressMap = async (newAddressMap: IAddressMapping): Promise<boolean> => {
         try {
+            const userAddressMap = newAddressMap;
             const clientMapping: any = this._clientMapping;
             const csAddressMap: any = {};
-            for (const key of Object.keys(newAddressMap)) {
-                csAddressMap[clientMapping[key]] = newAddressMap[key];
+            for (let walletCurrencySymbol of Object.keys(userAddressMap)) {
+                userAddressMap[walletCurrencySymbol.toLowerCase()] = userAddressMap[walletCurrencySymbol];
+                walletCurrencySymbol = walletCurrencySymbol.toLowerCase();
+                csAddressMap[clientMapping[walletCurrencySymbol]] = userAddressMap[walletCurrencySymbol];
```

Comment:
@SanchayPahalwani add an else branch to handle missing walletCurrencySymbol in clientMapping. Also add a negative test case.

Reply:
Done